test/test_compiler.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+import unittest
+import sys
+
+from stone.backends.python_types import (
+    PythonTypesBackend
+)
+from stone.compiler import (
+    Compiler
+)
+from stone.frontend.frontend import (
+    specs_to_ir
+)
+
+class PreserveBackend(PythonTypesBackend):
```

Comment:
Can we inherit from `CodeBackend` to make the tests language-independent?

Reply:
Currently the test cases using these classes are expecting a Python module to be used with `pytest` (so there's some coupling to a language here). I think changing the end-to-end test mentioned above to modular unit tests will remove the need for these / this module anyways.